platform/openide.util.lookup/src/org/openide/util/lookup/ProxyLookup.java

```diff
@@ -89,7 +110,96 @@ public synchronized String toString() {
         }
         return map.keySet();
     }
-    
+
+    /**
+     * A controller which allows the set of lookups being proxied to be
+     * set dynamically.
+     *
+     * @since 8.43
+     */
+    public static final class Controller {
+
+        private BiConsumer<? super Executor, ? super Lookup[]> consumer;
```

Comment:
I really want this change to not rely on JDK8 APIs. 

I suggest to change this field to `ProxyLookup`.